### ISSUE TITLE
Add code coverage to CI (+ a badge)

### ISSUE THREAD

**Issue by tasn** [svix/svix-webhooks#330]
Multiple options, not sure what's the right one:
- https://github.com/xd009642/tarpaulin
- https://github.com/mozilla/grcov
- llvm-cov as recommended in https://blog.rust-lang.org/inside-rust/2020/11/12/source-based-code-coverage.html

**Comment by michaelgrigoryan25** [comment]
What about using a GitHub Action? I think I found something at https://github.com/marketplace/actions/rust-grcov which uses grcov under the hood.

**Comment by tasn** [comment]
Thanks for the feedback! Yeah, it looks promising. We also want to have something that works locally, and we want to merge it with the test suite runs, but I'm sure we can get it working.

**Comment by michaelgrigoryan25** [comment]
In that case it may be possible to come up with something custom. cargo has [built-in support](https://doc.rust-lang.org/book/ch14-05-extending-cargo.html) for extending its functionality.

**Comment by tasn** [comment]
Yeah. I just hope [tarpaulin](https://github.com/xd009642/tarpaulin) will prove itself as the easy solution.
